### PR TITLE
Hb 4

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

File: src/main/java/com/cinema/dao/CinemaHallDao.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.dao;
+
+import com.cinema.model.CinemaHall;
+import java.util.List;
+
+public interface CinemaHallDao {
+
```
Comment (fyarkiy) [inline src/main/java/com/cinema/dao/CinemaHallDao.java]: it seems to me you can delete this empty line:)

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
```
Comment (fyarkiy) [inline src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]: save or persist?

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
```
Comment (fyarkiy) [inline src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: persist

File: src/main/java/com/cinema/model/MovieSession.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@Entity
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @NonNull
+    private Movie movie;
+    @ManyToOne
```
Comment (fyarkiy) [inline src/main/java/com/cinema/model/MovieSession.java:28]: By default - fetch type.Eager, i think. 
You need to push fetch type.Lazy...

File: src/main/java/com/cinema/model/MovieSession.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+@Data
+@RequiredArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
```
Comment (fyarkiy) [inline src/main/java/com/cinema/model/MovieSession.java:25]: FetchTyp = Eager. You need to push Lazy.

**Review by fyarkiy — APPROVED** [review]
Great work!

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 3 above)
Comment (ilya449) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: Thanks

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 2 above)
Comment (ilya449) [reply src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]: Persist, ty

File: src/main/java/com/cinema/model/MovieSession.java (hunk 5 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/MovieSession.java:25]: Why? It seems to me that Eager here is better...

File: src/main/java/com/cinema/dao/MovieSessionDao.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.dao;
+
+import com.cinema.model.MovieSession;
+import java.time.LocalDate;
+import java.util.List;
+
+public interface MovieSessionDao {
+    List<MovieSession> findSessions(Long movieId, LocalDate date);
```
Comment (Sofasmile) [inline src/main/java/com/cinema/dao/MovieSessionDao.java]:
```suggestion
    List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
```

File: src/main/java/com/cinema/model/CinemaHall.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+@RequiredArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "cinema_halls")
+public class CinemaHall {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NonNull
+    private int capacity;
```
Comment (Sofasmile) [inline src/main/java/com/cinema/model/CinemaHall.java]: can this field be null?

File: src/main/java/com/cinema/model/MovieSession.java (hunk 5 above)
Comment (Sofasmile) [reply src/main/java/com/cinema/model/MovieSession.java:25]: Для даного завдання це є опціональним. Головне пам'ятати що по дефолту @ ManyToOne має FetchTyp = Eager

File: src/main/java/com/cinema/model/CinemaHall.java (hunk 7 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/CinemaHall.java]: No, I wanted this field to be Integer

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**